Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,446 @@
+        {
+            if (m_gemSystemName->lineEdit()->text().isEmpty())
+            {
+                gemSystemNameIsValid = false;
+                m_gemSystemName->setErrorLabelText(tr("A gem system name is required."));
+            }
+
+            m_gemSystemName->setErrorLabelVisible(!gemSystemNameIsValid);
+        }
+        return gemSystemNameIsValid;
+
+    }
+
+    bool CreateAGemScreen::ValidateLicenseName()
+    {
+        bool licenseNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_license->lineEdit()->text().isEmpty())
+            {
+                licenseNameIsValid = false;
+                m_license->setErrorLabelText(tr("License details are required."));
+            }
+
+            m_license->setErrorLabelVisible(!licenseNameIsValid);
+        }
+        return licenseNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateCreatorName()
+    {
+        bool creatorNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            m_numNextClicks += 1;
+            if (m_numNextClicks > 1)
+            {
+                if (m_creatorName->lineEdit()->text().isEmpty())
+                {
+                    creatorNameIsValid = false;
+                    m_creatorName->setErrorLabelText(tr("A creator's name is required."));
+                }
+
+                m_creatorName->setErrorLabelVisible(!creatorNameIsValid);
+            }
+        }
+        return creatorNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateRepositoryURL()
+    {
+        bool repositoryURLIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_numNextClicks > 1)
+            {
+                if (m_repositoryURL->lineEdit()->text().isEmpty())
+                {
+                    repositoryURLIsValid = false;
+                    m_repositoryURL->setErrorLabelText(tr("A repository URL is required."));
+                }
+
+                m_repositoryURL->setErrorLabelVisible(!repositoryURLIsValid);
+            }
+        }
+        return repositoryURLIsValid;
+    }
+
+    void CreateAGemScreen::OnGemDisplayNameUpdated()
+    {
+        ValidateGemDisplayName();
+    }
+
+    void CreateAGemScreen::OnGemSystemNameUpdated()
+    {
+        ValidateGemSystemName();
+    }
+
+    void CreateAGemScreen::OnLicenseNameUpdated()
+    {
+        ValidateLicenseName();
+    }
+
+    void CreateAGemScreen::OnCreatorNameUpdated()
+    {
+        ValidateCreatorName();
+    }
+
+    void CreateAGemScreen::OnRepositoryURLUpdated()
+    {
+        ValidateRepositoryURL();
+    }
+
+    void CreateAGemScreen::HandleBackButton()
+    {
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            m_tabWidget->setCurrentIndex(1);
+            m_nextButton->setText("Next");
+        }
+        else if (m_tabWidget->currentIndex() == 1)
+        {
+            m_tabWidget->setCurrentIndex(0);
+        }
```

Comment:
```suggestion
        if (m_tabWidget->currentIndex() > 0)
        {
            m_tabWidget->setCurrentIndex(m_tabWidget->currentIndex() - 1 );
            ConvertNextButtonToCreate()
        }
```

Reply:
This function has since been modified with code that is specific to each branch, leaving this as is.